src/libcamera/mapped_framebuffer.cpp

```diff
@@ -223,12 +223,18 @@ MappedFrameBuffer::MappedFrameBuffer(const FrameBuffer *buffer, MapFlags flags)
 		const int fd = plane.fd.get();
 		auto &info = mappedBuffers[fd];
 		if (!info.address) {
+
+			LOG(Buffer, Debug) << "MappedFrameBuffer mmap plane: fd=" << fd
+						   << " length=" << info.mapLength
+						   << " flags=" << mmapFlags;
+
```

Comment:
This maps in a memory location accessible to the CPU the `destination` buffer where the encoder will write the JPEG encoded data. The `destination` buffer has been allocated by the android framework and the mapping seems to be successfull, indicating the issue is on the `source` buffer instead (the one allocated by the HAL as the source of the JPEG encoding operations)